sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -93,6 +93,21 @@ public boolean requiresDeduping() {
     return false;
   }
 
+  /**
+   * If offsetBasedDeduplicationEnabled returns true, then the UnboundedSource needs to provide the
+   * following:
+   *
+   * <ul>
+   *   <li>UnboundedReader which provides offsets that are unique for each element and
+   *       lexicographically ordered.
+   *   <li>CheckpointMark which provides an offset greater than all elements read and less than or
+   *       equal to the next offset that will be read.
+   * </ul>
+   */
+  public boolean offsetBasedDeduplicationEnabled() {
```

Comment:
sorry to still  nit on this, but Enabled could be confusing if the runner doesn't support it or use the offsets.

How about offsetBasedDeduplicationSupported? update the execution context method as well

Reply:
Done.